### PR TITLE
Update `chip_test.dart` tests for Material 3

### PR DESCRIPTION
**Pull request by TahaTesser**
Updated unit tests for `Tooltip` to have M2 and M3 versions.

More info in #139076

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/chip_test.dart [hunk 1]
```diff
@@ -1585,10 +2229,93 @@ void main() {
...
+      selectable: true,
+    );
+    await tester.pumpAndSettle();
+
+    expect(SchedulerBinding.instance.transientCallbackCount, equals(0));
+
+    // Simulate a tap on the label to select the chip.
+    await tester.tap(find.byKey(labelKey));
+    expect(selected, equals(true));
+    expect(SchedulerBinding.instance.transientCallbackCount, equals(isCanvasKit ? 3 : 1));
```
Comment (QuncCccccc) [inline packages/flutter/test/material/chip_test.dart:2280]: There are some test failures like this:
Screenshot 2024-01-04 at 12 50 24 PM

I think we need to add `kIsWeb` check before `isCanvasKit`.
```suggestion
    expect(SchedulerBinding.instance.transientCallbackCount, equals(kIsWeb && isCanvasKit ? 3 : 1));
```

Comment (TahaTesser) [reply packages/flutter/test/material/chip_test.dart]: You're right, i missed that! Thanks

File: packages/flutter/test/material/chip_test.dart [hunk 2]
```diff
@@ -3329,6 +4768,28 @@ void main() {
     expect(decoration.shape, shape);
   });
 
+  testWidgets('Material3 - Chip background color and shape are drawn on Ink', (WidgetTester tester) async {
+    const Color backgroundColor = Color(0xff00ff00);
+    const OutlinedBorder shape = ContinuousRectangleBorder();
+    final ThemeData theme = ThemeData();
```
Comment (QuncCccccc) [inline packages/flutter/test/material/chip_test.dart:4774]: nit: maybe it would be better if we can assign this theme to wrapForChip:)

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks for continuing to do this!

And I triaged all goldens:)

**Pull request merged**